indexer/services/l1/bridge/portal.go

```diff
@@ -0,0 +1,60 @@
+package bridge
+
+import (
+	"context"
+
+	"github.com/ethereum-optimism/optimism/indexer/db"
+	"github.com/ethereum-optimism/optimism/op-bindings/bindings"
+	"github.com/ethereum-optimism/optimism/op-service/backoff"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type Portal struct {
+	address  common.Address
+	contract *bindings.OptimismPortal
+}
+
+func NewPortal(addr common.Address, portal *bindings.OptimismPortal) *Portal {
```

Comment:
Do you need to pass in the bindings here? Can't you just pass in the address and client and then create the bindings inside of this? Because creating the bindings needs the address, it feels like a strange api

Reply:
The erc20 constructor follows the pattern noted above